Hard to understand error "no provider exists with the given name"

I was trying to run `yor` for testing on one of our dev applications in our terraform monorepo. When I ran it I got the following error which doesn't really indicate what I can do to fix it. Eg. which provider is it having issues with, etc.

```
❯ yor tag --directory envs/cetus/receipts
2021/05/27 14:00:16 [ERROR] failed to install provider for directory envs/cetus/receipts because of errors no provider exists with the given name
```

--- a/src/terraform/structure/terraform_module.go
+++ b/src/terraform/structure/terraform_module.go
@@ -63,11 +63,12 @@ func (t *TerraformModule) InitProvider() {
 		logger.MuteLogging()
 		_, diagnostics, err := providerInstaller.Get(pty, constraints.Versions)
 		logger.UnmuteLogging()
-		if diagnostics != nil && diagnostics.HasErrors() {
-			logger.Error(fmt.Sprintf("failed to install provider for directory %s because of errors %s", t.rootDir, diagnostics.Err()))
-		}
-		if err != nil {
-			logger.Error(fmt.Sprintf("failed to install provider for directory %s because of errors %s", t.rootDir, err))
+		if (diagnostics != nil && diagnostics.HasErrors()) || err != nil {
+			errMsg := diagnostics.Err()
+			if errMsg == nil {
+				errMsg = err
+			}
+			logger.Warning(fmt.Sprintf("failed to install provider \"%v\" for directory %s because of errors %s", provider, t.rootDir, errMsg))
 		}
 	}
 }

--- a/src/terraform/structure/terraform_module_test.go
+++ b/src/terraform/structure/terraform_module_test.go
@@ -1,8 +1,11 @@
 package structure
 
 import (
+	"os"
+	"path/filepath"
 	"testing"
 
+	"github.com/bridgecrewio/yor/tests/utils"
 	"github.com/stretchr/testify/assert"
 )
 
@@ -33,5 +36,15 @@ func TestTerrraformModule(t *testing.T) {
 	t.Run("Test TF Registry Module logic", func(t *testing.T) {
 		isRegistry := isTerraformRegistryModule("terraform-aws-modules/security-group/aws")
 		assert.True(t, isRegistry)
+	})
+
+	t.Run("Handle unsupported providers gracefully", func(t *testing.T) {
+		currentDir, _ := os.Getwd()
+		providersDir, _ := filepath.Abs(currentDir + "../../../../tests/terraform/providers")
+		output := utils.CaptureOutput(func() {
+			_ = NewTerraformModule(providersDir)
+		})
+
+		assert.Equal(t, "", output)
 	})
 }

--- a/src/terraform/structure/terraform_parser.go
+++ b/src/terraform/structure/terraform_parser.go
@@ -124,8 +124,9 @@ func (p *TerrraformParser) ParseFile(filePath string) ([]structure.IBlock, error
 		blockID := strings.Join(block.Labels(), ".")
 		terraformBlock, err := p.parseBlock(block, filePath)
 		if err != nil {
-			if strings.HasPrefix(err.Error(), "resource belongs to skipped provider") {
-				logger.Info(fmt.Sprintf("skipping block %s because the provider %s does not support tags", blockID, strings.Split(blockID, "_")[0]))
+			if strings.HasPrefix(err.Error(), "resource belongs to skipped") || strings.HasPrefix(err.Error(), "could not find client") {
+				logger.Info(fmt.Sprintf("skipping block %s because the provider %s does not exist locally or does not support tags",
+					blockID, strings.Split(blockID, "_")[0]))
 			} else {
 				logger.Warning(fmt.Sprintf("failed to parse terraform block because %s", err.Error()))
 			}

--- a/tests/terraform/providers/providers.tf
+++ b/tests/terraform/providers/providers.tf
@@ -1,0 +1,7 @@
+provider "sentry" {}
+
+resource "sentry_organization" "default" {
+  name        = "My Organization"
+  slug        = "my-organization"
+  agree_terms = true
+}

